services/web-app/utils/use-query-state.js

```diff
@@ -57,15 +28,35 @@ export const useQueryState = (
       return null
     }
 
-    const query = new URLSearchParams(window.location.search)
-    const queryValue = query.get(key)
-    const storageValue = saveToStorage ? localStorage.getItem(`${router.pathname}:${key}`) : null
+    const query = queryString.parseUrl(window.location.search, {
```

Comment:
If the query string changes programmatically from outside of this hook, is the expected behavior that the components using the `userQueryState` hook will be unaware of those changes?

Reply:
if the user changs it and presses enter it will 'cause a re-render, which will inform all comps listening for updates from use-query-state to be notified of that. if another component changes the queryState right now without going through the hook then yes, the components would be unaware but it's not an expected use-case for components to be manipulating the query state directly. 
Thoughts on this?

Reply:
Okay, got it. You're right that that's not an expected use case and is therefore pretty low priority. It could potentially be resolved by registering an event that listens for changes to the URL/query string and updating the hook accordingly, but no need to include that.